bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -250,29 +245,60 @@ public static double[] hsbToXY(HSBType hsb) {
      *
      * @param hsb an {@link HSBType} value.
      * @param gamut the color Gamut supported by the light.
-     * @return array of three double with the closest matching CIE 1931 x,y,Y in the range 0.0000 to 1.0000
+     * @return array of three or four double with the closest matching CIE 1931 x,y,Y in the range 0.0000 to 1.0000 -
+     *         plus an optional extra empty element to flag if the xyY result has been forced inside the given Gamut.
      */
     public static double[] hsbToXY(HSBType hsb, Gamut gamut) {
-        PercentType[] rgb = hsbToRgbPercent(hsb);
-        double r = inverseCompand(rgb[0].doubleValue() / PercentType.HUNDRED.doubleValue());
-        double g = inverseCompand(rgb[1].doubleValue() / PercentType.HUNDRED.doubleValue());
-        double b = inverseCompand(rgb[2].doubleValue() / PercentType.HUNDRED.doubleValue());
+        PercentType[] rgbPercents = hsbToRgbPercent(hsb);
+
+        // convert rgbPercents to doubles
+        double[] rgb = new double[rgbPercents.length];
```

Comment:
Why do you use an array here? I think that explicitely using `r`, `g`, `b` and `w` (like it was before) is much clearer.

Reply:
Ok. Done.